### DISCUSSION TITLE
how to query data thru Full-Text index in SQL/Cypher/Gremlin?

### DISCUSSION THREAD

**Discussion by frank-dkvan** [ArcadeData/arcadedb#452] [Q&A]
we have a type "Customer" has a text property "content"....and a  Full-Text index was created fro "content"

saying we want to find content ("hello, this is a new world" )

how to write SQL/Cypher/Gremlin scripts?  i.e. 
1.   select from Customer where content = ???????
2. {cypher} match (c : Customer) where c.content = ??????
3. {gremlin} g.V().has(.......)

**Comment by lvca** [comment]
You should be able to use the equals operator. When a full-text index is defined, the equals operator applies for full or partial match. Try `select from Customer where content = ?`

**Comment by frank-dkvan** [comment]
it works in SQL! thanks @lvca 

btw, how to make it in Cypher and Gremlin?  I tried the scripts below , but it doesn't work When a full-text index is defined
1.  {Cypher}  match (p) where p.content = 'you' return p
2. {Gremlin}  g.V().has('content',eq('you'))

**Comment by lvca** [comment]
Unfortunately Gremlin and Cypher queries are partially translated in SQL, not even all the times, so it won't work. I suggest to stick with SQL if you can or we should find an equivalent Gremlin operator for full-text searched we could support natively with a 1-1 bind.

**Reply by frank-dkvan** [reply]
I find JanusGraph supports  full-text search,which is built upon Gremlin and ElasticSearch/Apache Solar as Index backend.   since   ElasticSearch/Apache Solar  is essentially kind of distributed Lucene engine.    therefore, I am guessing we can find  an equivalent Gremlin operator to support full-text as well...
